packages/react-components/react-avatar/src/components/AvatarGroup/renderAvatarGroup.tsx

```diff
@@ -16,18 +16,16 @@ export const renderAvatarGroup_unstable = (state: AvatarGroupState) => {
     <AvatarGroupContext.Provider value={{ layout, size }}>
       <slots.root {...slotProps.root}>
         {state.root.children}
-        {state.hasOverflow && slots.overflowButton && slots.overflowSurface && slots.overflowList && (
+        {state.hasOverflow && slots.overflowButton && slots.overflowList && (
```

Comment:
overflowButton and overflowList are non-nullable, so this check shouldn't be required. (But it's possible that the typings don't work out correctly, so it's fine to leave these in if you get compile errors).
```suggestion
        {state.hasOverflow && (
```

Reply:
since I removed the partial and added the `?` in the types I need the check.

Reply:
Yeah I think it's a limitation of the types in ComponentState. They should never be undefined since the slots are NonNullable, but I think the ComponentState type doesn't get that right (and IIRC fixing it isn't straightforward). So it's fine to leave as-is.